docs/proposals/sysctl.md

```diff
@@ -0,0 +1,717 @@
+
+The node-level whitelist will be extensible via flags of the kubelet.
+
+### Pod API Changes
+
+Pod specification must be changed to allow the specification of kernel parameters:
+
+```go
+// Sysctl defines a kernel parameter to be set
+type Sysctl struct {
+	// Name of a property to set
+	Name string `json:"name"`
+	// Value of a property to set
+	Value string `json:"value"`
+}
+
+// PodSecurityContext holds pod-level security attributes and common container settings.
+// Some fields are also present in container.securityContext.  Field values of
+// container.securityContext take precedence over field values of PodSecurityContext.
+type PodSecurityContext struct {
+	...
+	// Sysctls hold a list of namespaced sysctls used for the pod. Pods with unsupported
+	// sysctls (by the container runtime) might fail to launch.
+	Sysctls []Sysctl `json:"sysctls,omitempty"`
+}
+```
+
+During alpha the extension of `PodSecurityContext` is modeled with annotations:
+
+```
+security.alpha.kubernetes.io/sysctls: kernel.shmmax:4,kernel.msgmax:1 2 3`
+```
+
+The value is a comma separated list of key-value pairs separated by colon.
+
+**Note**: none of the whitelisted (and in general none with the exceptions of descriptive plain text ones) sysctls use anything else than numbers, possibly separated with spaces.
+
+**Note**: sysctls must be on the pod level because containers in a pod share IPC and network namespaces (if pod.spec.hostIPC and pod.spec.hostNetwork is false) and therefore cannot have conflicting sysctl values. Moreover, note that all namespaced sysctl supported by Docker/RunC are either in the IPC or network namespace.
+
+### Apiserver Validation and Kubelet Admission
+
+#### In the Apiserver
+
+The name of each sysctl in `PodSecurityContext.Sysctls[*].Name` (or the `annotation security.alpha.kubernetes.io/sysctls` during alpha) is validated by the apiserver against:
+
+- 253 characters in length
+- it matches `sysctlRegexp`:
+
+```go
+const SysctlSegmentFmt string = "[a-z0-9]([_a-z0-9]*[a-z0-9])?"
+const SysctlFmt string = "(" + SysctlSegmentFmt + "\\.)*" + SysctlSegmentFmt
+var sysctlRegexp = regexp.MustCompile("^" + SysctlFmt + "$")
+```
+
+#### In the Kubelet
+
+The name of each sysctl in `PodSecurityContext.Sysctls[*].Name` (or the `annotation security.alpha.kubernetes.io/sysctls` during alpha) is checked by the kubelet against a static node-level *whitelist* of
+
+- specific sysctls
+- and a list of sysctl prefixes,
+
+These are defined under `pkg/kubelet` and to be maintained by the nodes team.
+
+The initial node-level whitelist will be:
+
+```go
+var whitelist = []string{
+    "kernel.shm*",
+    "net.ipv4.ip_local_port_range",
+    "net.ipv4.tcp_syncookies",
+    "net.ipv4.tcp_max_syn_backlog",
```

Comment:
`net.core.somaxconn` is one of the most wanted setting. Should we allow that anyways even though memory accounting is busted for it? I don't practically see the kernel fixing tcp accounting in the near future.